html_link_parser.py

```
@@ -0,0 +1,454 @@
+        # External links
+        if href.startswith(('http://', 'https://')):
+            try:
+                parsed = urlparse(link.href)
+                domain = parsed.netloc.lower()
+                
+                # Social media links
+                if any(social in domain for social in self.social_domains):
+                    return 'social'
+                    
+                return 'external'
+            except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)